### PR TITLE
Add support for removing a network by id

### LINKED ISSUE #1568: cannot delete network by ID
### Description

please check the title :-D!

### Steps to reproduce the issue

1.  `nerdctl network create deleteme --gateway 10.10.11.1 --subnet 10.10.11.0/24`
2.   `nerdctl network ls`
```
root@6518fb9254d2:/home/containerd/nerdctl# _output/nerdctl network ls
NETWORK ID      NAME        FILE
17f29b073143    bridge      /etc/cni/net.d/nerdctl-bridge.conflist
f9d6cec6deab    deleteme    /etc/cni/net.d/nerdctl-deleteme.conflist
                host        
                none  
````
4. `nerdctl network rm f9d6cec6deab`
````
FATA[0000] no such network: f9d6cec6deab
````

### Describe the results you received and expected

expectation:
delete network by ID.

### What version of nerdctl are you using?

[v1.0.0](https://github.com/containerd/nerdctl/releases/tag/v1.0.0)

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

_No response_

### Host information

_No response_

### PR DESCRIPTION
**Pull request by Mihai22125**
Resolves #1568 

First time contributor, so any feedback would be more than welcomed!

### CODE REVIEW

File: pkg/netutil/networkwalker.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+type NetworkWalker struct {
+	Client  *CNIEnv
+	OnFound OnFound
+}
+
+// Walk walks networks and calls w.OnFound .
+// Req is name, short ID, or long ID.
+// Returns the number of the found entries.
+func (w *NetworkWalker) Walk(ctx context.Context, req string) (int, error) {
+	longIDExp, _ := regexp.Compile(fmt.Sprintf("^sha256:%s.*", regexp.QuoteMeta(req)))
```
Comment (Zheaoli) [inline pkg/netutil/networkwalker.go:44]: The best practice here maybe not be to ignore errors here.

**Comment by Zheaoli** [comment]
And maybe we need extra docs change here to let people know they can delete network by id

File: pkg/netutil/networkwalker.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"fmt"
+	"regexp"
+)
+
+type Found struct {
+	Network      *networkConfig
+	Req          string // The raw request string. name, short ID, or long ID.
+	MatchIndex   int    // Begins with 0, up to MatchCount - 1.
+	MatchCount   int    // 1 on exact match. > 1 on ambiguous match. Never be <= 0.
+	UniqueImages int    // Number of unique images in all found images.
```
Comment (djdongjin) [inline pkg/netutil/networkwalker.go:30]: You can remove `UniqueImages` since it's not used and not relevant to network.

**Comment by Mihai22125** [comment]
Thanks for the review! I will address all comments soon.

**Review by djdongjin — COMMENTED** [review]
You may squash into 1 commit and sign.

CI will fail if you have any non-signed commits (https://github.com/containerd/nerdctl/pull/1581/commits/394e8173552ff07c209cb1ceaeba1969fb280a4c)

**Review by Zheaoli — APPROVED** [review]
LGTM on CI green

File: pkg/netutil/networkwalker.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
+/*
```
Comment (AkihiroSuda) [inline pkg/netutil/networkwalker.go:1]: Should this be in `pkg/idutil/netwalker` for consistency with other objects?
https://github.com/containerd/nerdctl/tree/main/pkg/idutil

Comment (Mihai22125) [reply pkg/netutil/networkwalker.go:1]: Hi @AkihiroSuda, I was thinking about this too, but `networkConfig` is not exported outside of `netutil` package, so the only solution I could figure out was to implement the walker in the same package as the struct.

Comment (Mihai22125) [reply pkg/netutil/networkwalker.go]: @AkihiroSuda @Zheaoli Addressed, please review

Comment (fahedouch) [reply pkg/netutil/networkwalker.go]: @Mihai22125  @Zheaoli   No LGTM for  public `networkConfig`. PTAL https://github.com/containerd/nerdctl/pull/1538#discussion_r1030761472 !

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**